presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java

```diff
@@ -36,4 +39,18 @@ private MoreMaps() {}
                 .flatMap(Collection::stream)
                 .collect(toMap(Map.Entry::getKey, Map.Entry::getValue, merger));
     }
+
+    public static <K, V> Map<K, V> immutableLinkedMap(Map<K, V> map)
```

Comment:
I double checked before asking that question, but now I see I've read the docs for `ImmutableMap.copyOf(Iterable)` - which says 'The returned map iterates over entries in the same order as the original iterable'. So, at the very least, we can `ImmutableMap.copyOf(map.entrySet())` in this method's body.

Also, the copyOf(Map) method's doc don't guarantee it to retain order, but it effectively does. So we can consider writing a learning test that will fail once they break the contract and use the copyOf.

> but I'd expect it'd be you, @ArturGajowy, who could object to that
not sure why though